torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -5829,6 +5829,29 @@ def test_owner_rref_forward_synchronization3(self):
     def test_owner_rref_forward_synchronization4(self):
         self._test_owner_rref_forward_synchronization("cuda:1", "cuda:1")
 
+    @skip_if_lt_x_gpu(1)
+    def test_tensor_view_as_args(self):
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options = self.rpc_backend_options
+        options.set_device_map(dst, {0 : 0})
+
+        rpc.init_rpc(
+            name=worker_name(self.rank),
+            backend=self.rpc_backend,
+            rank=self.rank,
+            world_size=self.world_size,
+            rpc_backend_options=options,
+        )
+
+        x = torch.randn(1000, 200).cuda(0)
+        futs = []
+        for xx in x.split(100):
+            futs.append(rpc.rpc_async(dst, RpcTest._identity, args=(xx,)))
```

Comment:
I don't see how this test triggered the bug: here there's only always one stream, which is always the current one, so the device guard we added shouldn't be having any effect. Are you sure this failed before? Could we make it clearer what we're testing for?

Also, I'd like us to stop relying on subtle timing races to catch these issues, it's too unreliable. If we intend to test that one operation waits for another one, we should explicitly delay the "earlier" operation by an insanely large amount (say, one second) using `torch.cuda._sleep` so that it becomes exceedingly unlikely that the "later" operation just casually happens to synchronize correctly. You can see how I did it in the tests for CUDAFuture just below.